crates/node/src/compiler.rs

```diff
@@ -0,0 +1,147 @@
+use alloy_primitives::{hex, Bytes, B256};
+use core::panic;
+use reth_primitives::Bytecode;
+use reth_revm::{
+    handler::register::EvmHandler,
+    interpreter::{InterpreterAction, SharedMemory, EMPTY_SHARED_MEMORY},
+    Context as RevmContext, Database, Frame,
+};
+use revmc::{primitives::SpecId, EvmCompilerFn, EvmLlvmBackend, OptimizationLevel};
+use std::{
+    collections::HashMap,
+    sync::{
+        mpsc::{Receiver, Sender},
+        Arc, Mutex,
+    },
+    thread,
+};
+
+#[derive(Debug)]
+pub struct ExternalContext {
+    sender: Sender<(B256, Bytes)>,
+    // TODO: cache shouldn't be here (and should definitely not be wrapped in a mutex)
+    cache: Arc<Mutex<HashMap<B256, EvmCompilerFn>>>,
+}
+
+impl ExternalContext {
+    pub fn new(spec_id: SpecId) -> Self {
+        let cache = Arc::new(Mutex::new(HashMap::new()));
+        let (sender, receiver) = std::sync::mpsc::channel();
+
+        // TODO: graceful shutdown
+        thread::spawn({
+            let cache = cache.clone();
+
+            move || {
+                let context = Box::leak(Box::new(revmc::llvm::Context::create()));
```

Comment:
Can I leak this, does that make sense?  I guess I shouldn't need to since `context` will live for the whole `while` loop below.

Reply:
You should be able to use `with_llvm_context` and nest the implementation inside of the closure